### PR TITLE
DBZ-7451 Support for standard BSON representation of UUID in incremental snapshots

### PR DESCRIPTION
**Pull request by jcechace**
https://issues.redhat.com/browse/DBZ-7451

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -638,6 +640,19 @@ private Object[] keyFromRow(Object[] row) {
             case STRING:
                 key = documentId.asString().getValue();
                 break;
+            case BINARY:
+                var subtype = documentId.asBinary().getType();
+                if (!BsonBinarySubType.isUuid(subtype)) {
+                    throw new IllegalStateException("Unsupported type of document id");
```
Comment (jpechane) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java:646]: @jcechace Would it be possible to handle unrecognized binary as either `byte[]` or String-encoded byte value?

Comment (jcechace) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java:646]: In theory yes, however I'm not sure how well that's gonna play with our snapshotting logic (e.g. determining the max/min key values and so on).

Theoretically we could even consider ignoring UUID and just treat everything as binary

Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java:646]: Fair point, we cannot make any assumptions around orderin so it is better to handle it case by case.

**Comment by jpechane** [comment]
@jcechace Applied, thanks

**Pull request merged**